pepper-apis/src/main/java/org/broadinstitute/ddp/db/dao/ValidationDao.java

```diff
@@ -140,15 +140,9 @@ default void insertValidations(QuestionDef question, long revisionId) {
         I18nContentRenderer i18nContentRenderer = new I18nContentRenderer();
 
         for (RuleDto ruleDto : ruleDtos) {
-            String correctionHint = null;
-            if (ruleDto.getHintTemplateId() != null) {
-                correctionHint = i18nContentRenderer.renderContent(getHandle(), ruleDto.getHintTemplateId(), langCodeId);
+            if (ruleDto.getQuestionId() == questionDto.getId()) {
```

Comment:
This `if` condition here means the list of `RuleDto`s might not belong to the question. I think this is a red flag that something is wrong here... `getAllActiveValidations(questionDto)` should only return rules for the question!

Reply:
The idea was that it's flat list with all rules of all questions of whole activity.